wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/exception/InstallerExceptionMapper.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.installer.server.exception;
+
+import org.eclipse.che.api.core.rest.shared.dto.ServiceError;
+import org.eclipse.che.dto.server.DtoFactory;
+
+import javax.inject.Singleton;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @author andrew00x
+ * @author gazarenkov
+ */
+@Provider
+@Singleton
+public class InstallerExceptionMapper implements ExceptionMapper<InstallerException> {
+    @Override
+    public Response toResponse(InstallerException exception) {
+        ServiceError serviceError = DtoFactory.newDto(ServiceError.class).withMessage(exception.getMessage());
+        String errorAsJson = DtoFactory.getInstance().toJson(serviceError);
+
+        if (exception instanceof InstallerNotFoundException) {
+            return Response.status(Response.Status.NOT_FOUND)
+                           .entity(errorAsJson)
+                           .type(MediaType.APPLICATION_JSON)
+                           .build();
+
+        } else if (exception instanceof InstallerConflictException) {
+            return Response.status(Response.Status.CONFLICT)
+                           .entity(errorAsJson)
+                           .type(MediaType.APPLICATION_JSON)
+                           .build();
+
+        } else if (exception instanceof IllegalInstallerKeyException) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
I think 403 forbidden should be used instead as server understood request and entity is valid but value in entity prevents fulfilment of the request